### ISSUE TITLE
About the unsupervised loss and the contrast loss.

### ISSUE THREAD

**Issue by DISAPPEARED13** [Haochen-Wang409/U2PL#67]
Hi, there!
Sorry to bother again!
after run training with your code in my project, I found that my unsupervised loss and contrast loss not decrease but keep fluctuating around a number, what's the trend of unsupervised loss and contrast loss going, when in your training process?
Thanks a lot! @Haochen-Wang409 

Yours sincerely,
D.

**Comment by DISAPPEARED13** [comment]
And in the algorithm 1 of the paper, we can get probabilities and representations by different output heads, 
So in algorithm 1, gradient backward like this:
f->h, and g->h. h update for 2 times, and f, g update 1 time.
It that right?
But what if my network just a straight out situation, 
and i extract a feature map from the middle of the network, that means my network will be updated for 2 times, 
how the gradient update?
I doubt that my unsupervised loss and contrast loss not decrease because that...? Or just the ratio?

**Comment by Haochen-Wang409** [comment]
Hi, the contrastive loss will not decrease significantly since it is more likely to be a regularizer than the optmization objective. Unsupervised loss is expected to decrease as the training procedure goes. But it may be not so stable as supervised loss since the noisy in pseudo-labels.

**Comment by DISAPPEARED13** [comment]
I thought for days, the contrast loss is calculated based on a updating ratio of unrealiable pixels, this may still cause some information to learn, so you inspired me that it play a regularization role while optimizing the subjects.
While for unsupervised loss, do you mean during the training progress, the noise is from the under-fitting teacher model's instability?

**Comment by Haochen-Wang409** [comment]
Yes, even for **reliable** pseudo-labels for unlabeled data might be wrong, since they are just predictions from the teacher model.
